ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/WebClientOptionsFactory.java

```diff
@@ -24,20 +24,52 @@
 import java.util.Optional;
 
 import io.vertx.core.net.PfxOptions;
+import io.vertx.core.net.ProxyOptions;
 import io.vertx.ext.web.client.WebClientOptions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 class WebClientOptionsFactory {
+  private static final Logger LOG = LogManager.getLogger();
+
   public WebClientOptions createWebClientOptions(final Config config) {
     final WebClientOptions clientOptions =
         new WebClientOptions()
             .setDefaultPort(config.getDownstreamHttpPort())
             .setDefaultHost(config.getDownstreamHttpHost())
-            .setTryUseCompression(true);
+            .setTryUseCompression(true)
+            .setProxyOptions(proxyOptions().orElse(null));
 
     applyTlsOptions(clientOptions, config);
     return clientOptions;
   }
 
+  private static Optional<ProxyOptions> proxyOptions() {
+    var proxyHost = System.getProperty("http.proxyHost", "none");
+    if ("none".equalsIgnoreCase(proxyHost)) {
+      return Optional.empty();
+    }
+    int proxyPort;
+    try {
+      proxyPort = Integer.valueOf(System.getProperty("http.proxyPort", "80"));
```

Comment:
If you `Integer.parseInt` you can avoid using an Integer and autoboxing back to int.